### PR TITLE
Fix gazebo catkin warning

### PR DESCRIPTION
**Pull request by davetcoleman**
Same fix as https://github.com/ros-simulation/gazebo_ros_pkgs/pull/521 except for ``gazebo_ros_control`` pkg

*This PR is sponsored by [Vicarious](http://www.vicarious.com/)*

### CODE REVIEW

**Comment by davetcoleman** [comment]
I'm not 100% this PR is good, I'm getting errors in downstream packages (baxter_simluation):

```
tried to find library '-lpthread'.  The
  library is neither a target nor built/installed properly.
```

**Comment by j-rivero** [comment]
> I'm not 100% this PR is good, I'm getting errors in downstream packages (baxter_simluation):

sounds to me like one of the dependencies is exporting that pthread in an invalid way. Maybe looking into the CMakeCache.txt file can give you hint or try to disable the sdformat/boost find_package lines alternatively to see which one is the culprit.

File: gazebo_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -75,8 +75,6 @@ link_directories(
   ${catkin_LIBRARY_DIRS}
 )
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
```
Comment (davetcoleman) [inline gazebo_plugins/CMakeLists.txt:78]: this line is a duplicate from previous in this file

File: gazebo_ros/CMakeLists.txt [hunk 2]
```diff
@@ -36,21 +36,20 @@ generate_dynamic_reconfigure_options(cfg/Physics.cfg)
```
Comment (davetcoleman) [inline gazebo_ros/CMakeLists.txt:36]: In this file I only do OCD house keeping :)

File: gazebo_ros_control/CMakeLists.txt [hunk 3]
```diff
@@ -31,9 +33,11 @@ catkin_package(
     joint_limits_interface
     urdf
     angles
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME} default_robot_hw_sim
-  DEPENDS gazebo
```
Comment (davetcoleman) [inline gazebo_ros_control/CMakeLists.txt:36]: From my testing, ``DEPENDS gazebo`` caused the issues I was having

**Comment by davetcoleman** [comment]
This PR is ready for review/merging

File: gazebo_ros_control/CMakeLists.txt (hunk 3 above)
Comment (j-rivero) [reply gazebo_ros_control/CMakeLists.txt:36]: According to [catkin documentation](http://wiki.ros.org/catkin/CMakeLists.txt) that `DEPENDS` clause is fine (at least to my eyes). My vote is for keeping it and debug where is exactly the error.

Comment (davetcoleman) [reply gazebo_ros_control/CMakeLists.txt:36]: I read that documentation and it does not clarify anything for me. It just says ``DEPENDS - Non-catkin CMake projects that this project depends on`` but then why does it also not include things like Boost?

Another example - why does ``gazebo_ros`` not [DEPEND on GAZEBO](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/kinetic-devel/gazebo_ros/CMakeLists.txt#L55)? My change in this PR was inspired by that CMakeLists file.

Comment (j-rivero) [reply gazebo_ros_control/CMakeLists.txt:36]: > It just says DEPENDS - Non-catkin CMake projects that this project depends on but then why does it also not include things like Boost?
> 
> Another example - why does gazebo_ros not DEPEND on GAZEBO?

I would say that both are bugs or errors in the configuration of our CMakeLists.txt. I found this [answer](http://answers.ros.org/question/58498/what-is-the-purpose-of-catkin_depends/) specially useful to understand what are they used for. Now it makes sense that changes in `DEPEND` will affect third party software.

Comment (davetcoleman) [reply gazebo_ros_control/CMakeLists.txt:36]: That's a really good answer, would be great if it was added to the catkin documentation. I added a link from the wiki to that answer, for now.

Seems I'll have the same pthread linking issues now, I'll work on it soon.

Comment (davetcoleman) [reply gazebo_ros_control/CMakeLists.txt:36]: Ok, I've spent the last hour on the issue and I think our problem is with Gazebo, see https://bitbucket.org/osrf/gazebo/issues/2198/gazebo7-linking-error-with-cmake-catkin

Comment (j-rivero) [reply gazebo_ros_control/CMakeLists.txt:36]: given the current state of the issue, I'm ok for not declaring `gazebo` as `DEPENDS` here. Dave, if you could please leave the line commented with a comment pointing to the gazebo issue that should be enough to restore it as soon as we find the way of solving the issue. After the comments, good to merge.

Comment (davetcoleman) [reply gazebo_ros_control/CMakeLists.txt:36]: i've added the comment

**Comment by davetcoleman** [comment]
ping @j-rivero, ready for merge

**Comment by j-rivero** [comment]
> ping @j-rivero, ready for merge

Thanks Dave!

**Pull request merged**